core/src/main/java/org/elasticsearch/search/sort/SortBuilder.java

```diff
@@ -55,34 +54,24 @@
     protected SortOrder order = SortOrder.ASC;
     public static final ParseField ORDER_FIELD = new ParseField("order");
 
-    private static final Map<String, SortBuilder<?>> PARSERS;
+    private static final Map<String, Parser<?>> PARSERS;
+
+    @FunctionalInterface
+    public interface Parser<T extends SortBuilder<?>> {
```

Comment:
Doing it. I'm going to move it to the bottom of the file so it isn't so much in the way.